mailbox/api/src/test/java/org/apache/james/mailbox/mock/DataProvisioner.java

```diff
@@ -0,0 +1,112 @@
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.mock;
+
+import java.io.UnsupportedEncodingException;
+
+import javax.mail.Flags;
+
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.MailboxConstants;
+import org.apache.james.mailbox.model.MailboxPath;
+
+public class DataProvisioner {
+    
+    /**
+     * Number of Domains to be created in the Mailbox Manager.
+     */
+    public static final int DOMAIN_COUNT = 3;
+    
+    /**
+     * Number of Users (with INBOX) to be created in the Mailbox Manager.
+     */
+    public static final int USER_COUNT = 3;
+    
+    /**
+     * Number of Sub Mailboxes (mailbox in INBOX) to be created in the Mailbox Manager.
+     */
+    public static final int SUB_MAILBOXES_COUNT = 3;
+    
+    /**
+     * Number of Sub Sub Mailboxes (mailbox in a mailbox under INBOX) to be created in the Mailbox Manager.
+     */
+    public static final int SUB_SUB_MAILBOXES_COUNT = 3;
+    
+    /**
+     * The expected Mailboxes count calculated based on the feeded mails.
+     */
+    public static final int EXPECTED_MAILBOXES_COUNT = DOMAIN_COUNT * 
+                     (USER_COUNT + // INBOX
+                      USER_COUNT * SUB_MAILBOXES_COUNT + // INBOX.SUB_FOLDER
+                      USER_COUNT * SUB_MAILBOXES_COUNT * SUB_SUB_MAILBOXES_COUNT);  // INBOX.SUB_FOLDER.SUBSUB_FOLDER
+    
+    /**
+     * Number of Messages per Mailbox to be created in the Mailbox Manager.
+     */
+    public static final int MESSAGE_PER_MAILBOX_COUNT = 3;
+    
+    /**
+     * Utility method to feed the Mailbox Manager with a number of 
+     * mailboxes and messages per mailbox.
+     */
+    public static void feedMailboxManager(MailboxManager mailboxManager) throws MailboxException, UnsupportedEncodingException {
+        for (int i = 0; i < DOMAIN_COUNT; i++) {
```

Comment:
No way to use `IntStream.range(0, DOMAIN_COUNT)`?

Reply:
I guess it's to avoid sneakyThrow()

Reply:
No it's just that this code is uggly, I did not wanted to spend too much time reworking it...

Reply:
Here is the alternative, but look not better :))
```java
        IntStream.range(0, DOMAIN_COUNT).boxed()
            .flatMap(domainCount -> IntStream
                .range(0, USER_COUNT)
                .mapToObj(userCount -> Pair.of(domainCount, userCount)))
            .forEach(Throwing
                .<Pair<Integer, Integer>>consumer(pair -> provisionUser(mailboxManager, "user" + pair.getKey() + "@localhost" + pair.getValue()))
                .sneakyThrow());
```